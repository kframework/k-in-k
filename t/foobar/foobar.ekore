module A-BASIC-K
  sort K{} []
  sort KItem{} []
endmodule [ ]

module B-KSEQ
  import A-BASIC-K []

  symbol kseq{}(KItem{}, K{}) : K{} []
  symbol append{}(K{}, K{}) : K{} [function{}()]
  symbol dotk{}() : K{} []

  axiom{ R }
    \equals{K{},R}(
      append{}(dotk{}(),K2:K{}),
      K2:K{})
  []

  axiom{R}
    \equals{K{},R}(
      append{}(kseq{}(K1:KItem{},K2:K{}),K3:K{}),
      kseq{}(K1:KItem{},append{}(K2:K{},K3:K{})))
  []
endmodule [ ]

module C-INJ
  symbol inj{From,To}(From) : To []

  axiom{S1,S2,S3,R}
    \equals{S3,R}(
      inj{S2,S3}(inj{S1,S2}(T:S1)),
      inj{S1,S3}(T:S1))
  []
endmodule

module D-K
  import B-KSEQ []
  import C-INJ []
endmodule

module FOOBAR
  import D-K []
  syntax Foo ::=
      "bar" [ klabel ( bar   ) , function ]
    | "buzz" [ klabel ( buzz2   ) ]
  rule bar {  } (  ) => buzz2 {  } (  )
  syntax Fizz ::=
    "foobar" [ klabel ( foobar  ) ]
endmodule
