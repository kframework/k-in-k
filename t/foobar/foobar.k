module FOOBAR
  syntax Layout ::= r"(/\\*([^\\*]|(\\*+([^\\*/])))*\\*+/|//[^\n\r]*|[\\ \n\r\t])*" [klabel(layout)]

  syntax Pgm ::= Foo
  syntax Foo ::= "bar" [klabel(bar), function]
               | "buzz" [klabel(buzz)]
  rule bar => buzz
endmodule
