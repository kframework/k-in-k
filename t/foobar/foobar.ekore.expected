[  ]

module FOOBAR
  sort Buzz {  } [  ]
  symbol bar {  } (  ) : Foo {  } [ function {  } (  )  ]
  symbol buzz2 {  } (  ) : Foo {  } [  ]
  syntax Foo ::=
    "bar" [ klabel ( bar   ) , function ]
    | "buzz" [ klabel ( buzz2   ) ]
  sort Foo {  } [  ]
  sort Fizz {  } [  ]
  syntax Fizz ::=
    "foobar" [ klabel ( foobar  ) ]
  symbol foobar {  } (  ) : Fizz {  } [  ]
  rule bar {  } (  ) => buzz2 {  } (  )
  axiom{ R } \equals{ Foo { } , R } ( bar {  } (  )  , buzz2 {  } (  ) ) [ ]

endmodule [  ]

